### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by asdf2014** [apache/incubator-superset#2245]
superset==0.15.4

When i import the dashboard, i meet the problem:
![image](https://cloud.githubusercontent.com/assets/8108788/23296205/fa7b078c-faae-11e6-8b13-39e7b3e9efbe.png)

**Comment by asdf2014** [comment]
Should we consider the situation rather than throw a `500 - Internal Server Error`?
```
if db_ds is None or len(db_ds) == 0:
     return None
return db_ds[0]
```
![image](https://cloud.githubusercontent.com/assets/8108788/23296288/4d237fd2-faaf-11e6-8c17-5f07a6fd4c83.png)

**Comment by xrmx** [comment]
A more pythonic way should be this:
```
if not db_ds
   return None
```

need to check if the callers handle None or nor.

**Comment by asdf2014** [comment]
For me, it looks still terrible.

![image](https://cloud.githubusercontent.com/assets/8108788/23296516/6372e024-fab0-11e6-9d76-7014ce6cf1cc.png)

**Comment by asdf2014** [comment]
@xrmx Nice! U r so great.

**Comment by asdf2014** [comment]
Then i try to change the name of druid cluster into the same as the other `superset` cluster...

![image](https://cloud.githubusercontent.com/assets/8108788/23296777/97e059b2-fab1-11e6-9ae6-df0d3c33ce15.png)

**Comment by asdf2014** [comment]
That means i should edit the all `metrics`, `slices` and `dashboard` that involved with the druid cluster?

**Comment by asdf2014** [comment]
I used the __VIOLENT__ way...
```
alter table datasources drop FOREIGN KEY `datasources_ibfk_2`;
update clusters set cluster_name='Druid Cluster' where cluster_name='druid cluster';
update datasources set cluster_name ='Druid Cluster' where cluster_name ='druid cluster';
alter table datasources add constraint  `datasources_ibfk_2`  FOREIGN KEY (`cluster_name`) REFERENCES `clusters` (`cluster_name`);
```
### troubleshooting
    show create table datasources; 
![image](https://cloud.githubusercontent.com/assets/8108788/23299241/c5aab7ac-fabb-11e6-83ab-2fb2ca977324.png)
![image](https://cloud.githubusercontent.com/assets/8108788/23299265/e12cf4c2-fabb-11e6-827b-e4d1b3b397f2.png)

**Comment by asdf2014** [comment]
How can we gracefully solve this problem? @mistercrunch

**Comment by xrmx** [comment]
@asdf2014 FYI flooding maintainers emails with tens of updates per day and naming them just because you want to have your issues fixed is not cool.

**Comment by asdf2014** [comment]
Alright... :(

**Comment by xrmx** [comment]
@asdf2014 it's cool you experiment with superset but github issues are not a chat :)

**Comment by asdf2014** [comment]
@xrmx Can you just focus on sloving problem？！

**Comment by mistercrunch** [comment]
Notice: this issue has been closed because it has been inactive for 422 days. Feel free to comment and request for this issue to be reopened.

**Comment by ZisisFl** [comment]
I'm facing the same issue. I exported 2 data sources (the demo main and one postgresql) and the related dashboards from a Superset venv of mine. Then I created and other one got it working added the 2 data sources successfully checked the connection and also created some random charts. Then I tried to import the dashboards, the only one that was successfully imported was from recently imported demo db of flights. All dashboards that are connected to the posgresql database give me this error when I try to load them. I have double checked that eveything has the same name, no changes in columns etc.
